src/decode.rs

```diff
@@ -1446,8 +1446,12 @@ unsafe fn decode_b_inner(
     let has_chroma = f.cur.p.layout != Rav1dPixelLayout::I400
         && (bw4 > ss_hor || t.bx & 1 != 0)
         && (bh4 > ss_ver || t.by & 1 != 0);
-
-    let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
+    let is_inter_or_switch = f
+        .frame_hdr
+        .as_ref()
+        .unwrap()
+        .frame_type
+        .is_inter_or_switch();
```

Comment:
Could you just copy out `frame_type`?  Since `frame_type` is used elsewhere, too, and is `Copy`.  It'd probably help more to avoid other borrow checker errors in the future vs. just doing `is_inter_or_switch`.

Reply:
Sure that also works. I also updated one other place that was using `is_inter_or_switch`.

Reply:
It looks there's still borrow checker errors, from another use of `frame_hdr.frame_type`, so `Copy`ing out the whole `frame_type` would fix that I think.  Also, there seem to be unused import warnings so if you want to fix these here, too, feel free to.